### PR TITLE
Fix for Sqlproj tree bug

### PR DESCRIPTION
**Pull request by Benjin**
Fixes bug where files located in a subfolder defined before the folders themselves in .sqlproj would result in those files not appearing in the ADS treeview

### CODE REVIEW

File: extensions/sql-database-projects/src/models/tree/fileFolderTreeItem.ts [hunk 1]
```diff
@@ -75,7 +92,6 @@ function fsPathToProjectUri(fileSystemUri: vscode.Uri, projectNode: ProjectRootT
 		localUri = fileSystemUri.fsPath.substring(projBaseDir.length);
 	}
 	else {
-		vscode.window.showErrorMessage('Project pointing to file outside of directory');
```
Comment (kisantia) [inline extensions/sql-database-projects/src/models/tree/fileFolderTreeItem.ts:78]: was this removed on purpose?

File: extensions/sql-database-projects/src/models/tree/projectTreeItem.ts [hunk 2]
```diff
@@ -53,6 +46,10 @@ export class ProjectRootTreeItem extends BaseProjectTreeItem {
 		for (const entry of this.project.files) {
 			const parentNode = this.getEntryParentNode(entry);
 
+			if (Object.keys(parentNode.fileChildren).includes(entry.fsUri.path)) {
+				continue; // ignore duplicate entries
```
Comment (kisantia) [inline extensions/sql-database-projects/src/models/tree/projectTreeItem.ts:50]: add a test for ignoring duplicate entries

File: extensions/sql-database-projects/src/models/tree/fileFolderTreeItem.ts (hunk 1 above)
Comment (Benjin) [reply extensions/sql-database-projects/src/models/tree/fileFolderTreeItem.ts:78]: Yup, whatever's calling this should be handling displaying errors to the user.

File: extensions/sql-database-projects/src/test/projectTree.test.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import * as should from 'should';
+import * as vscode from 'vscode';
+
+import { Project } from '../models/project';
+import { FolderNode, FileNode, sortFileFolderNodes } from '../models/tree/fileFolderTreeItem';
+import { ProjectRootTreeItem } from '../models/tree/projectTreeItem';
+
+describe('Project Tree tests', function (): void {
+	it('Should correctly order tree nodes by type, then by name', async function (): Promise<void> {
```
Comment (SakshiS-harma) [inline extensions/sql-database-projects/src/test/projectTree.test.ts:14]: Can we have one more test with deeper file structure, to verify the functionality works as expected?

Comment (Benjin) [reply extensions/sql-database-projects/src/test/projectTree.test.ts:14]: Since VSCode requests the children of a node per-node rather than all at once, only one layer of children can be sorted at a time.  I'll make a task so I remember to add tests for a deep tree itself.

Comment (SakshiS-harma) [reply extensions/sql-database-projects/src/test/projectTree.test.ts:14]: Sounds good.
I have one question here for my own understanding: the ordering here. This tree ordering is for the opened project in Explorer, right? So, what is the behavior right now for- SchemaObjectType, where the tree would be deeper than one level?

Comment (Benjin) [reply extensions/sql-database-projects/src/test/projectTree.test.ts:14]: Actually, I kinda killed two birds with one stone with the new test case, since it also confirms ordering at all depths!
